apis/Makefile

```diff
@@ -1,12 +1,115 @@
+# Description: Makefile for generating code from protobuf files for Seldon Core
+#
+#	By default, this Makefile will download the protoc compiler and any required
+#	language-specific plugins locally, in the `tools` directory. The versions of
+#	the downloaded tools are pinned to what is specified in the `VERSION SELECTION`
+#	section below.
+#
+# Overwrite the PROTOC_GEN_*_VERSION variables in that section to use different
+# releases.
+#
+# If you want to use versions of protoc and/or of the language-specific plugins
+# already installed on your system, you can do so by overwriting the
+# CMD_PROTOC* variables. This is not recommended because it offers significantly
+# fewer guarantees about the compatibility of the generated code across the
+# codebase
+#
+
 .PHONY: build
-build: build-go build-kotlin build-v2-python
+build: build-go build-kotlin build-v2-python update-copyright
+
+### VERSION SELECTION ###
+
+# Links in comments below point to the corresponding projects, for determining
+# the latest version of protoc + language-specific plugins.
+
+# https://github.com/protocolbuffers/protobuf/releases
+PROTOC_VERSION ?= 21.10
+
+# https://pkg.go.dev/google.golang.org/protobuf/cmd/protoc-gen-go
+PROTOC_GEN_GO_VERSION ?= 1.28.1
+
+# https://pkg.go.dev/google.golang.org/grpc/cmd/protoc-gen-go-grpc
+PROTOC_GEN_GO_GRPC_VERSION ?= 1.2.0
+
+# https://pypi.org/project/grpcio-tools/
+PROTOC_GEN_PYTHON_GRPCIO_TOOLS_VERSION ?= 1.51.3
+
+# https://github.com/grpc/grpc-java
+# https://mvnrepository.com/artifact/io.grpc/protoc-gen-grpc-java
+PROTOC_GEN_JAVA_GRPC_VERSION ?= 1.45.1
+
+# https://github.com/grpc/grpc-kotlin
+# https://mvnrepository.com/artifact/io.grpc/protoc-gen-grpc-kotlin
+PROTOC_GEN_KOTLIN_GRPC_VERSION ?= 1.2.1
+PROTOC_GEN_KOTLIN_GRPC_JDK ?= jdk7
+
+
+################################################################################
+
+MAKEFILE := $(lastword $(MAKEFILE_LIST))
+BASE_DIR := $(realpath $(dir $(MAKEFILE)))
+TOOLS_DIR := $(BASE_DIR)/tools
```

Comment:
nit: should this be `.tools` to make it explicit that these are not files to check in (we are guarded anyway with .gitignore)

Reply:
Indeed, I thought of that briefly as well. Will update